scanner/src/main/kotlin/experimental/ExperimentalScanner.kt

```diff
@@ -421,6 +428,51 @@ class ExperimentalScanner(
             ScanResult(provenance, scanner.details, summary)
         }
     }
+
+    private fun createFileArchives(nestedProvenances: Map<Package, NestedProvenance>) {
+        // TODO: The archives are currently created in a way compatible with the existing implementation in the
+        //       LocalScanner. This allows to keep using existing file archives without changing the logic used to
+        //       access those archives in the reporter. To achieve this nested provenances are downloaded recursively,
+        //       so that the created archives contain also files from nested repositories.
+        //       This could be replaced with creating file archives for each provenance separately and building the
+        //       final result on demand, to reduce duplication in the file archives.
+
+        val archiver = scannerConfig.archive.createFileArchiver()
+
+        val provenancesWithMissingArchives = nestedProvenances.filterNot { (_, nestedProvenance) ->
+            archiver.hasArchive(nestedProvenance.root)
+        }
+
+        log.info { "Creating file archives for ${provenancesWithMissingArchives.size} packages." }
+
+        val duration = measureTime {
+            provenancesWithMissingArchives.forEach { (_, nestedProvenance) ->
+                val dir = downloadRecursively(nestedProvenance)
+                archiver.archive(dir, nestedProvenance.root)
+                dir.safeDeleteRecursively(force = true)
+            }
+        }
+
+        log.info {
+            "Created file archives for ${provenancesWithMissingArchives.size} packages in " +
+                    "${duration.inWholeMilliseconds} ms."
+        }
+    }
+
+    private fun downloadRecursively(nestedProvenance: NestedProvenance): File {
+        // Use the provenanceDownloader to download each provenance from nestedProvenance separately, because they are
+        // likely already cached if a local scanner wrapper is used.
+
+        val root = provenanceDownloader.download(nestedProvenance.root)
+
+        nestedProvenance.subRepositories.forEach { (path, provenance) ->
+            val tempDir = provenanceDownloader.download(provenance)
+            val targetDir = root.resolve(path)
+            tempDir.toPath().moveTo(targetDir.toPath())
```

Comment:
Can we rely on source and target files to reside on the same filesystem? If so, please pass `StandardCopyOption.ATOMIC_MOVE` as a copy option.